src/content/getusermedia/record/js/main.js

```diff
@@ -0,0 +1,144 @@
+/*
+ *  Copyright (c) 2014 The WebRTC project authors. All Rights Reserved.
+ *
+ *  Use of this source code is governed by a BSD-style license
+ *  that can be found in the LICENSE file in the root of the source
+ *  tree.
+ */
+
+'use strict';
+
+/* globals MediaRecorder */
+
+// This code is adapted from
+// https://rawgit.com/Miguelao/demos/master/mediarecorder.html
+
+var mediaSource = new MediaSource();
+mediaSource.addEventListener('sourceopen', handleSourceOpen, false);
+var mediaRecorder;
+var recordedBlobs;
+var sourceBuffer;
+
+navigator.getUserMedia = navigator.getUserMedia ||
```

Comment:
I thought about this, and then I decided that in this case `adapter.js` added more complexity than it saved.

However — I'll change it to use `navigator.mediaDevices` and Promises.